### PR TITLE
TD-664/Aws-cf-templates for release-4.0

### PR DESCRIPTION
**Pull request by ermirizio**
M4 template sync for TD-664 branch AWS CF templates

### CODE REVIEW

**Comment by ermirizio** [comment]
Not sure about this test is ok to leave it failing

CI tests / Go 1.15 Redis 5 (pull_request)

File: .goreleaser.yml [hunk 1]
```diff
@@ -193,6 +189,8 @@ dockers:
     - "tykio/tyk-plugin-compiler:{{ .Tag }}"
     - "tykio/tyk-plugin-compiler:v{{ .Major }}.{{ .Minor }}{{.Prerelease}}"
   build_flag_templates:
+    - "--build-arg=branch={{ .Branch }}"
```
Comment (alephnull) [inline .goreleaser.yml:192]: This line and 193 are not required.

**Review by alephnull — CHANGES_REQUESTED** [review]
Its a minor change but it needs to be changed in the templates.

File: .goreleaser.yml (hunk 1 above)
Comment (ermirizio) [reply .goreleaser.yml]: LINE 193:
- "--build-arg=TYK_GW_TAG={{ .Tag }}"

Plugin compiler image still relies on that ARG, so not gonna remove it.

**Pull request merged**